### PR TITLE
msgpack: disable C++ headers

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

### CODE REVIEW

**Comment by jonchang** [comment]
This doesn't seem right. The C library shouldn't be installing .hpp files in the first place, which is usually for C++ headers.

**Comment by cho-m** [comment]
Based on C/C++ separation, I am guessing this will be a temporary conflict until upstream does a C-library release. It may be worth having a comment to check if we can remove conflict in next `msgpack` release.

**Comment by jonchang** [comment]
We might be able to disable it by setting `MSGPACK_ENABLE_CXX` in the call to cmake.

File: Formula/msgpack.rb [hunk 1]
```diff
@@ -17,6 +17,8 @@ class Msgpack < Formula
 
   depends_on "cmake" => :build
 
+  conflicts_with "msgpack-cxx", because: "both install msgpack.hpp"
+
   def install
     system "cmake", ".", *std_cmake_args
```
Comment (jonchang) [inline Formula/msgpack.rb:23]:
```suggestion
    system "cmake", ".", *std_cmake_args, "-DMSGPACK_ENABLE_CXX=OFF"
```

**Comment by carlocab** [comment]
> We might be able to disable it by setting `MSGPACK_ENABLE_CXX` in the call to cmake.

This would break existing users though. I'd rather wait to hear what upstream has to say about it: https://github.com/msgpack/msgpack-c/issues/974

**Comment by jonchang** [comment]
The C library hasn't cut a 4.0 release yet, but the commit disabling building C++ headers by default is in the upstream `c_master` branch.

**Review by chenrui333 — APPROVED** [review]

**Pull request closed without merging**